src/main/java/org/mamute/auth/LDAPApi.java

```diff
@@ -116,6 +117,34 @@ public boolean authenticate(String username, String password) {
 	}
 
 	/**
+	 * Authenticates a user WITHOUT a password presented. The user name must
+	 * come from a trusted SSO source, e.g. the <code>getRemoteUser()</code>
+	 * from the HTTP Servlet Request.
+	 * 
+	 * @param username
+	 *            User name to authenticate in any case. If no matching user
+	 *            account exists, it is created.
+	 * 
+	 * @return <code>true</code> if the user could be found in LDAP and the User
+	 *         object was initialized, <code>false</code> otherwise.
+	 */
+	public boolean authenticateSSO(String username) {
+
+		try (LDAPResource ldap = new LDAPResource()) {
+			if (ldap.lookupUser(username) == null) {
```

Comment:
I'm a little confused as to why you need to also check LDAP for the user if the incoming source is trusted. Seems inefficient to me.